telegram-bot-api/StatsJson.h

```diff
@@ -0,0 +1,285 @@
+    object(td::Slice(BotStatActor::DESCR[1]), td::JsonLong(five_sec_));
+    object(td::Slice(BotStatActor::DESCR[2]), td::JsonLong(one_min_));
+    object(td::Slice(BotStatActor::DESCR[3]), td::JsonLong(one_hour_));
+  }
+ private:
+  const td::int64 inf_;
+  const td::int64 five_sec_;
+  const td::int64 one_min_;
+  const td::int64 one_hour_;
+};
+
+class JsonStatsBotStats : public td::Jsonable {
+ public:
+  explicit JsonStatsBotStats(td::vector<ServerBotStat> stats) : stats_(std::move(stats)) {
+    CHECK(BotStatActor::SIZE == 4 && "Check that we have 4 fields.");
+  }
+  void store(td::JsonValueScope *scope) const {
+    auto object = scope->enter_object();
+    object("request_count", JsonStatsBotStatDouble(stats_[0].request_count_, stats_[1].request_count_, stats_[2].request_count_, stats_[3].request_count_));
+    object("request_bytes", JsonStatsBotStatDouble(stats_[0].request_bytes_, stats_[1].request_bytes_, stats_[2].request_bytes_, stats_[3].request_bytes_));
+    object("request_file_count", JsonStatsBotStatDouble(stats_[0].request_file_count_, stats_[1].request_file_count_, stats_[2].request_file_count_, stats_[3].request_file_count_));
+    object("request_files_bytes", JsonStatsBotStatDouble(stats_[0].request_files_bytes_, stats_[1].request_files_bytes_, stats_[2].request_files_bytes_, stats_[3].request_files_bytes_));
+    object("request_files_max_bytes", JsonStatsBotStatLong(stats_[0].request_files_max_bytes_, stats_[1].request_files_max_bytes_, stats_[2].request_files_max_bytes_, stats_[3].request_files_max_bytes_));
+    object("response_count", JsonStatsBotStatDouble(stats_[0].response_count_, stats_[1].response_count_, stats_[2].response_count_, stats_[3].response_count_));
+    object("response_count_ok", JsonStatsBotStatDouble(stats_[0].response_count_ok_, stats_[1].response_count_ok_, stats_[2].response_count_ok_, stats_[3].response_count_ok_));
+    object("response_count_error", JsonStatsBotStatDouble(stats_[0].response_count_error_, stats_[1].response_count_error_, stats_[2].response_count_error_, stats_[3].response_count_error_));
+    object("response_bytes", JsonStatsBotStatDouble(stats_[0].response_bytes_, stats_[1].response_bytes_, stats_[2].response_bytes_, stats_[3].response_bytes_));
+    object("update_count", JsonStatsBotStatDouble(stats_[0].update_count_, stats_[1].update_count_, stats_[2].update_count_, stats_[3].update_count_));
+  }
+
+ protected:
+  const td::vector<ServerBotStat> stats_;
+};
+
+class JsonStatsBotAdvanced : public JsonStatsBot {
+ public:
+  explicit JsonStatsBotAdvanced(const std::pair<td::int64, td::uint64> score_id_pair,
+                                const ServerBotInfo bot,
+                                const td::vector<ServerBotStat> stats,
+                                const bool hide_sensible_data,
+                                const double now)
+      : JsonStatsBot(score_id_pair), bot_(bot), stats_(stats), hide_sensible_data_(hide_sensible_data), now_(now) {
```

Comment:
std::move for score_id_pair, bot and stats as you are already moving them from ClientManager

Reply:
Done